lib/pharos/phases/validate_hostname.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Pharos
+  module Phases
+    class ValidateHostname < Pharos::Phase
+      title "Validate unique hostnames for all hosts"
+
+      # @param hosts Array<Pharos::Configuration::Host>
```

Comment:
Phases already have `@config` so you could just use `@config.hosts` from the phase and ignore whatever is in `@host`.